extension/js/common/browser/browser-msg.ts

```diff
@@ -321,7 +321,10 @@ export class BrowserMsg {
       BrowserMsg.sendCatch(dest, 'pgp_block_render', bm);
     },
     pgpBlockReady: (frame: ChildFrame, bm: Bm.PgpBlockReady) => {
-      BrowserMsg.sendToParentWindow(frame, 'pgp_block_ready', bm);
+      Catch.setHandledTimeout(() => {
+        // Adding a timeout here because the parent frame might not be ready immediately to receive the event
+        BrowserMsg.sendToParentWindow(frame, 'pgp_block_ready', bm);
+      }, 2000);
```

Comment:
@sosnovsky, I've identified the reason some messages are getting stuck during loading.

The issue arises because sometimes `PgpBlockView.setHandlers` is called before `Inbox.setHandlers`/`setup-webmail-content-script.browserMsgListen`.
As a result, when `PgpBlockView.setHandlers` tries to send `BrowserMsg.send.pgpBlockReady`, `Inbox.setHandlers` isn't set up to listen to this event yet.
I know adding a 2-second static delay seems like a makeshift solution, and I'm not entirely convinced it's the best approach. I'm looking into alternative solutions now.

What are your thoughts? Also, could you test this PR to see if you can replicate the loading issue?


https://github.com/FlowCrypt/flowcrypt-browser/blob/2b2c2599c9258457aa2ac8da7fff8ce46cb8ed9f/extension/chrome/settings/inbox/inbox.ts#L100
https://github.com/FlowCrypt/flowcrypt-browser/blob/9fa013af570f6d74846b622fc0a344cece3d1d2f/extension/chrome/elements/pgp_block.ts#L68
https://github.com/FlowCrypt/flowcrypt-browser/blob/65387ffdb475e38efa3ed82005052d92ccfc1339/extension/js/content_scripts/webmail/generic/setup-webmail-content-script.ts#L225

Reply:
This 2-second delay seems to fix issue for me too, however it makes loading of all messages quite long.
It'll be great to find some way for checking if `browserMsgListen` and `setHandlers` were performed before sending messages, or maybe retrying message sending if it wasn't handled.

Interesting why initial issue happens only in `encrypted inbox` and in firefox gmail page, and works well in chrome gmail page. Can be some differences in message sending logic.